api/gateway/share_test.go

```diff
@@ -1,123 +1,45 @@
 package gateway
 
 import (
-	"bytes"
 	"testing"
 
-	"github.com/stretchr/testify/assert"
-
 	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/celestia-app/pkg/namespace"
+	"github.com/celestiaorg/celestia-app/pkg/shares"
+	"github.com/stretchr/testify/require"
+	coretypes "github.com/tendermint/tendermint/types"
 )
 
 func Test_dataFromShares(t *testing.T) {
-	type testCase struct {
-		name    string
-		input   [][]byte
-		want    [][]byte
-		wantErr bool
+	input := [][]byte{
+		[]byte("beep"),
+		[]byte("beeap"),
+		[]byte("BEEEEAHP"),
 	}
 
-	smallTxInput := padShare([]uint8{
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, // namespace id
-		0x1,                // info byte
-		0x0, 0x0, 0x0, 0x2, // 1 byte (unit) + 1 byte (unit length) = 2 bytes sequence length
-		0x0, 0x0, 0x0, 17, // reserved bytes
-		0x1, // unit length of first transaction
-		0xa, // data of first transaction
-	})
-	smallTxData := []byte{0x1, 0xa}
-
-	largeTxInput := [][]byte{
-		fillShare([]uint8{
-			0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, // namespace id
-			0x1,                // info byte
-			0x0, 0x0, 0x2, 0x2, // 512 (unit) + 2 (unit length) = 514 sequence length
-			0x0, 0x0, 0x0, 17, // reserved bytes
-			128, 4, // unit length of transaction is 512
-		}, 0xc), // data of transaction
-		padShare(append([]uint8{
-			0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, // namespace id
-			0x0,                // info byte
-			0x0, 0x0, 0x0, 0x0, // reserved bytes
-		}, bytes.Repeat([]byte{0xc}, 19)..., // continuation data of transaction
-		)),
+	ns := namespace.RandomBlobNamespace()
+	sss := shares.NewSparseShareSplitter()
+	for _, i := range input {
+		b := coretypes.Blob{
+			Data:             i,
```

Comment:
optional
Confused this with index, might worth having a better name here. data?